timelock-impl/src/main/java/com/palantir/atlasdb/timelock/corruption/LocalCorruptionHandler.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.corruption;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.rholder.retry.RetryException;
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.timelock.corruption.TimeLockCorruptionNotifier;
+import com.palantir.tokens.auth.AuthHeader;
+
+public class LocalCorruptionHandler {
+    private static final Logger log = LoggerFactory.getLogger(
+            LocalCorruptionHandler.class);
+
+    private static final AuthHeader AUTH_HEADER = AuthHeader.valueOf("Bearer omitted");
+    private final List<TimeLockCorruptionNotifier> corruptionNotifiers;
+    private final Retryer<Void> corruptionNotifyRetryer;
+    private final ExecutorService executorService;
+
+    public LocalCorruptionHandler(
+            List<TimeLockCorruptionNotifier> corruptionNotifiers) {
+        this.corruptionNotifiers = corruptionNotifiers;
+        this.executorService = PTExecutors.newFixedThreadPool(corruptionNotifiers.size());
```

Comment:
This is a bit too resource intensive now, as we are always creating this executor service, even when no corruption happened (also you forgot to shut it down eventually).
Also there is no need to do this asynchronously at this point really, so just run it synchronously

Reply:
Ah right, that makes sense.